### DISCUSSION TITLE
Why does hostname gets prefixed with lima-<hostname> and why are .local lookups not working?

### DISCUSSION THREAD

**Discussion by BitWise-0x** [lima-vm/lima#1634] [Q&A] [answered]
These are two questions in one but somewhat related.,

1.-  The instance names always get prefixed with lima-<hostname>
  ` limactl list` shows the instance name I picked verbatim, and I would like the instance to have that very name.   The instance however, say I do a lookup from the instance itself against its IP, always returns the lima-   .  So does `hostname`

```
nslookup 10.0.1.7
7.1.0.10.in-addr.arpa	name = lima-GALAXY-DOCK.
7.1.0.10.in-addr.arpa	name = lima-GALAXY-DOCK.local.
```

2.- Regarding .local resolution, I understand it is using a local agent to pull DNS via the host resolver, however, it isn't working.  `nslookups` or `dig`  against other systems in my network (non lima guests as well as physical, etc return a failure in name resolution.   Do I need to install AVAHI on the LiMa instance for this to work fully?

I am running on bridge mode, getting an IP from my DHCP just fine, and so I am basically trying to make it so I don't have to use /etc/hosts anywhere in my network and I just access my lima instances via HOSTNAME.LOCAL and viceversa.  I come from Multipass and I always installed AVAHI on all instances there but it seemed like LiMa does a much better job at controlling how DNS and networking works which is great - Multipass is a bit of a wild mess for stuff like this.   

**UPDATE on (2)**:  I just installed AVAHI on one instance and .local resolution seems to be working.  However, I am still curious as to that is even  necessary with LiMa out of the box...?

**Comment by a-h** [comment]
There's some related information about local resolution at https://github.com/lima-vm/lima/discussions/2018

**Comment by AkihiroSuda** [comment, marked as answer]
> 1.- The instance names always get prefixed with lima-
 limactl list shows the instance name I picked verbatim, and I would like the instance to have that very name. The instance however, say I do a lookup from the instance itself against its IP, always returns the lima- . So does hostname

Because I wanted to name the default instance "default" but I didn't want a host named "default" to exist.
Also, hostnames like "ubuntu", "debian", etc. are likely to conflict with existing hosts.

If somebody submits a PR to allow specifying a custom hostname, I think we can happily accept it.

> UPDATE on (2): I just installed AVAHI on one instance and .local resolution seems to be working. However, I am still curious as to that is even necessary with LiMa out of the box...?

I guess this should just corresponds to the distro (Ubuntu)'s choice?

**Reply by BitWise-0x** [reply]
makes sense...  I just hostname it so it isnt worth the effort, thanks for the follow up....

**Comment by aborrero** [comment]
running `hostname` inside the VM after creation is not enough, given things like DNS entries and /etc/hosts will remain unchanged.

To properly remove the `lima` prefix, I think we need a way to set the VM name at creation time.
